embed_grader/serapis/views/courses.py

```diff
@@ -192,61 +192,55 @@ def download_csv(request, course_id):
 
     writer.writerow(headers)
 
-    assignment_ids = []
-
-    for assign in assignment_list:
-        assignment_ids.append(assign.id)
-
-    print(assignment_ids)
-
     students = [o.user_fk for o in CourseUserList.objects.filter(course_fk=course)]
+    students = [s for s in students if not s.has_perm('modify_course', course)]
+
     scores = {}
-    for s in students:
-        scores[s] = [0. for _ in range(len(assignment_list))]
-    for idx, aid in enumerate(assignment_ids):
-        assignment = Assignment.objects.get(id=aid)
+    max_possible_score_for_each_assignment = [0.0]*(len(assignment_list))
+    for student in students:
+        scores[student] = [0. for _ in range(len(assignment_list))]
+    for idx, aid in enumerate(assignment_list):
+        assignment = Assignment.objects.get(id=aid.id)
         teams = Team.objects.filter(assignment_fk=assignment)
         for team in teams:
             last_submission = grading.get_last_submission(team, assignment)
         if last_submission is None:
             continue
-        _, score, _ = last_submission.retrieve_task_grading_status_and_score_sum(True)
+        _, score, max_possible_score_for_each_assignment[idx] = last_submission.retrieve_task_grading_status_and_score_sum(True)
         team_students = [o.user_fk for o in TeamMember.objects.filter(team_fk=team)]
-        for s in team_students:
-            if s.id != 1:
-                scores[s][idx] = score
+        for student in students:
+            if student.id != 1:
+                # the score of student s in the idx-th assignment
+                # idx is the index in the assignment list
+                scores[student][idx] = score
 
-    for s in scores:
+    for score in scores:
```

Comment:
Sorry but I think I was wrong.... `score` should be `student`... Please update it